flink-core/src/main/java/org/apache/flink/api/java/typeutils/PojoTypeInfo.java

```diff
@@ -393,6 +443,18 @@ public void addComparatorField(int fieldId, TypeComparator<?> comparator) {
 				keyFields.size() == fieldComparators.size(),
 				"Number of key fields and field comparators is not equal.");
 
+			Tuple2<ArrayList<Integer>, Class> custCompKey = new Tuple2(keyFieldIds, getTypeClass());
+			if (customComparators.containsKey(custCompKey)) {
+				return InstantiationUtil.instantiate(customComparators.get(custCompKey));
+			}
+
+			if (config.isCodeGenerationEnabled()) {
+				return new PojoComparatorGenerator<T>(keyFields.toArray(new Field[keyFields.size()]),
+					fieldComparators.toArray(new TypeComparator[fieldComparators.size()]), createSerializer
+					(config), getTypeClass(), keyFieldIds.toArray(new Integer[keyFields.size()]), config)
```

Comment:
I think it's less verbose to use the parameterless toArray, and cast to the appropriate array type.